src/msg/Dispatcher.h

```diff
@@ -18,9 +18,9 @@
 
 #include "include/buffer_fwd.h"
 #include "include/assert.h"
+#include "Message.h"
 
 class Messenger;
-class Message;
```

Comment:
maybe just duplicate the existing MessageRef typedef here, so you can still use the forward declaration of Message? for that reason, i prefer MessageRef to the new nested Message::ref/const_ref

Reply:
Moved MessageRef to `MessageRef.h`. What do you think now @cbodley ?

Reply:
that works! in the seastar messenger, i added a [net/Fwd.h](https://github.com/ceph/ceph/blob/master/src/crimson/net/Fwd.h) with forward declarations for all of the messenger types. so i could see this living in a more general header, but it's up to you